File: made_in_python/bogo_sort.py
from random import randint
from time import time
from random import shuffle

the_list = [0] * 11 # Array size that bigger than 11 takes a looong time to sort.

for i in range(len(the_list)): # Python's integer is 64 bit. Becuz of that I made the randomizer like this.
    the_list[i] = randint(-2147483648, 2147483647 - 1) # -1 becuz the endpoint is included

def sort(number_list):
    """ In Bogo Sort; firstly, the program checks if the array is sorted. If not, it shuffels the array and
    checks again if the array is sorted. The program repeates this process until the array is sorted. """

    i: int
    b: bool = True

    for i in range(1, len(number_list)): # Cheking if the array is sorted
        if number_list[i - 1] > number_list[i]:
            b = False
            break

    while not b:
        b = True
        shuffle(number_list)

        for i in range(1, len(number_list)): # Cheking if the array is sorted
            if number_list[i - 1] > number_list[i]:
                b = False
                break

start_time: float = time()
sort(the_list)
end_time: float = time()
#print("[" + ", ".join(str(x) for x in the_list) + "]") # To see the array
print((str)(len(the_list)) + " integers sorted in " + (str)((end_time - start_time) * 1000) + " milliseconds.")

File: MadeInJava/src/BogoSort.java
import java.util.Random;
//import java.util.Arrays; // Use this too if you want to see the array

public class BogoSort {
    public static void main(String[] args) {
        int[] theArray = new int[11]; // Array size that bigger than 11 takes a looong time to sort.
        Random r = new Random();

        for (int i = 0; i < theArray.length; i++)
        {
            theArray[i] = r.nextInt();
        }

        long startTime = System.nanoTime();
        Sort(theArray);
        long endTime = System.nanoTime();
        //System.out.println(Arrays.toString(theArray)); // To see the array
        System.out.println(theArray.length + " integers sorted in " + ((endTime - startTime) / 1000000.0) + " milliseconds");
    }

    static void Sort(int[] numberArray)
    {
        /* In Bogo Sort; firstly, the program checks if the array is sorted. If not, it shuffels the array and
        checks again if the array is sorted. It continues this pattern until the array is sorted. */

        int i;
        boolean b = true;

        for (i = 1; i < numberArray.length; i++) // Cheking if the array is sorted
        {
            if (numberArray[i - 1] > numberArray[i])
            {
                b = false;
                break;
            }
        }

        if (b)
        {
            return;
        }

        int[] bogoSortedArray = new int[numberArray.length];
        int temp;
        Random r = new Random();

        while (!b)
        {
            i = 0;
            b = true;
            boolean[] didIUseThisElementForSelectingRandomlyArray = new boolean[numberArray.length];

            while (i < numberArray.length)
            {// I couldn't find a built in primitive integer array shuffler. So, here is my shuffler.
                temp = r.nextInt(numberArray.length);

                if (!didIUseThisElementForSelectingRandomlyArray[temp])
                {
                    bogoSortedArray[i++] = numberArray[temp]; // This means bogoSortedArray[i] = numberArray[temp]; i += 1;
                    didIUseThisElementForSelectingRandomlyArray[temp] = true;
                }
            }

            for (i = 1; i < numberArray.length; i++) // Cheking if the array is sorted
            {
                if (bogoSortedArray[i - 1] > bogoSortedArray[i])
                {
                    b = false;
                    break;
                }
            }
        }

        for (i = 0; i < numberArray.length; i++)
        {
            numberArray[i] = bogoSortedArray[i];
        }
    }
}
